src/NLog/Config/LoggingConfiguration.cs

```diff
@@ -285,7 +289,9 @@ public TTarget FindTargetByName<TTarget>(string name)
         }
 
         /// <summary>
-        /// Add a rule with min- and maxLevel.
+        /// Add a rule with <paramref name="minLevel"/> and <paramref name="maxLevel"/>. The rule will NOT have a name.<br/>
```

Comment:
The LoggingRule-name is an exotic feature, that is only relevant for special users that wants to special changes to NLog-configuration at runtime. Think normal users will wonder what is the meaning of rule-name? Is a rule-name needed? What happens if no rule-name is provided? I prefer that xml-comments doesn't mention all the stuff that are not assigned.

Reply:
In our current implementation of logging in our software, which uses .NET's `TraceSource`, the operator may change the logging levels of all classes (i.e. of each class that contains a `TraceSource`) at any time. Logging has become an essential part in out software, because it's not a "normal" windows desktop application, but an application that controls a complex production machine. We may not simply shut down the application and restart with different logging levels. Also, logging is so important because there are not thousands or millions of copies of our software, but just a few hundred. And each copy has a different configuration because of different machine setup. If a failure occurs in a customer's machine, we take the logs from this machine and solve the problem within hours. These logs must have tons of information in order to find the problem's cause immediately. And this requires a highly configurable logging system, otherwise our log files would reach hundreds of megabytes within minutes.
These are the reasons that make the name of the logging rule very relevant for us. We will configure NLog almost always only programmatically (default logging level `warning` for all loggers) and not via config file, although there may be a config file to set up an initial configuration (to log details if something goes wrong during machine startup).

Reply:
I'm happy that you like the flexibility of the NLog-configuration. And maybe the xml-comments for the `RuleName`-property could be improved to highlight the abilities it provides. But I still prefer that one doesn't try to mention all the stuff that are not assigned by the constructor.